rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -92,33 +93,128 @@ void SequentialCompressionWriter::setup_compression()
     throw std::invalid_argument{
             "SequentialCompressionWriter requires a CompressionMode that is not NONE!"};
   }
-  compressor_ = compression_factory_->create_compressor(compression_options_.compression_format);
+
+  setup_compressor_threads();
+}
+
+void SequentialCompressionWriter::setup_compressor_threads()
+{
+  ROSBAG2_COMPRESSION_LOG_DEBUG(
+    "setup_compressor_threads: Starting %lu threads",
+    compression_options_.compression_threads);
+  if (compression_options_.compression_threads < 1) {
+    // This should have already been set to something reasonable prior to this
+    // point, but we'll double-check just to make sure.
+    auto hardware_threads = std::thread::hardware_concurrency();
+    compression_options_.compression_threads = hardware_threads > 0 ? hardware_threads : 1;
+  }
+  compression_is_running_ = true;
+  auto compress_fn = [&]()
```

Comment:
Stylistically - i think I would prefer if this compression function was not specified inside `setup_compressor_threads` - it's a little long and makes the method harder to read. Nitpick though, take or leave it.

Reply:
I don't have a preference either way, so I'll go ahead and move it out.